backend/app/agent/agent_director/llm_agent_director.py

```diff
@@ -111,13 +114,19 @@ async def execute(self, user_input: AgentInput) -> AgentOutput:
         :return: The output from the agent
         """
         try:
+            if self._state is None:
+                raise RuntimeError("AgentDirectorState must be set before executing")
             first_call: bool = True
             transitioned_to_new_phase: bool = False
             agent_output: AgentOutput | None = None
             while first_call or transitioned_to_new_phase:
                 if self._state.current_phase == ConversationPhase.ENDED:
+                    try:
```

Comment:
TBC.

instead of always doing.

```python
try:
   finished_msg = t("messages", "agent_director.final_message")
except Exception:
    finished_msg = "The conversation has finished!"
```

I would suggest.

```python
t("messages", "agent_director.final_message", fallback_message="The conversation has finished!")
```